base/iterators.jl

```diff
@@ -649,6 +649,52 @@ end
 iterate(it::Drop, state) = iterate(it.xs, state)
 isdone(it::Drop, state) = isdone(it.xs, state)
 
+
+# takewhile
+
+takewhile(pred::Function,xs) = TakeWhile(xs,pred)
```

Comment:
I would prefer defining the struct first; that's the style used by the rest of the file.